server/src/main/java/org/elasticsearch/index/query/support/NestedScope.java

```diff
@@ -37,11 +46,28 @@ public NestedObjectMapper nextLevel(NestedObjectMapper level) {
         return previous;
     }
 
+    /**
+     * @param innerHitBuilder The inner hit builder to set as current inner hit builder
+     * @return The previous inner hit builder
+     */
+    public InnerHitBuilder nextLevelInnerHits(InnerHitBuilder innerHitBuilder) {
```

Comment:
I needed a way to tell the top-level & knn Query how many child documents we are wanting to gather. Since kNN is eager and returns only a wrapper around `ScoreDoc` values, we need to know how many total documents we care about.